app/components/post_list/post/body/content/permalink_preview/index.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
+import {of as of$} from 'rxjs';
+import {map, switchMap, distinctUntilChanged} from 'rxjs/operators';
+
+import {getDisplayNamePreferenceAsBool} from '@helpers/api/preference';
+import {observePost} from '@queries/servers/post';
+import {queryDisplayNamePreferences} from '@queries/servers/preference';
+import {observeUser, observeTeammateNameDisplay, observeCurrentUser} from '@queries/servers/user';
+
+import PermalinkPreview from './permalink_preview';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+const enhance = withObservables(['embedData'], ({database, embedData}: WithDatabaseArgs & {embedData: PermalinkEmbedData}) => {
+    const teammateNameDisplay = observeTeammateNameDisplay(database);
+    const currentUser = observeCurrentUser(database);
+
+    const preferences = queryDisplayNamePreferences(database).observeWithColumns(['value']);
+    const isMilitaryTime = preferences.pipe(map((prefs) => getDisplayNamePreferenceAsBool(prefs, 'use_military_time')));
```

Comment:
Maybe we should rename the function DisplayNamePreferences to DisplayPreferences? what do you think? cause it caught me off guard for a minute, had to double check the helpers.